packages/block/src/block.ts

```diff
@@ -299,11 +316,76 @@ export class Block {
     }
   }
 
-  private _validateUncleHeader(uncleHeader: BlockHeader, blockchain: Blockchain) {
-    // TODO: Validate that the uncle header hasn't been included in the blockchain yet.
-    // This is not possible in ethereumjs-blockchain since this PR was merged:
-    // https://github.com/ethereumjs/ethereumjs-blockchain/pull/47
-    const height = this.header.number
-    return uncleHeader.validate(blockchain, height)
+  private async _validateUncleHeaders(uncleHeaders: BlockHeader[], blockchain: Blockchain) {
+    if (uncleHeaders.length == 0) {
+      return
+    }
+
+    // validate the header
+    for (let i = 0; i < uncleHeaders.length; i++) {
+      const header = uncleHeaders[i]
```

Comment:
Also not a blocker, but I find `for (const uncleHeader of uncleHeaders)` more elegant and simple.

Reply:
I did all the others without a loop and instead with a `map`, don't know why I used a for loop here. Note that `for(const x of y)` is something that ESLint does not like (we have ignored that rule). I don't know why ESLint doesn't like that syntax.

Reply:
Ah right, I can't directly use `map` because it calls an `async` function.

Reply:
Have rewritten this using `map` and `Promise.all`, personally, I like it 😄